src/app/profile/yourGiving/historicalView/historicalView.component.js

```diff
@@ -34,11 +40,56 @@ class HistoricalView {
       this.loadingGiftsError = true
     })
   }
+
+  isRecent (year, month) {
+    const today = moment()
+    return today.subtract(3, 'months').isBefore(moment(`${year}/${month}/01`, 'YYYY/MM/DD'))
+  }
+
+  parseHistoricalGifts (historicalGifts, year, month) {
+    if (!historicalGifts.length || !historicalGifts[0].donations) {
+      return historicalGifts
+    }
+
+    const filteredList = historicalGifts.filter((historicalGift) => {
+      historicalGift.donations = historicalGift.donations.filter((donation) => {
```

Comment:
This mutates the `donations` attributes on `historicalGifts` items. But that should be OK because it's not used again after being passed to `parseHistoricalGifts`, right?

Reply:
I believe that to be true, though can make it immutable if it is better practice.

Reply:
In general, I try to avoid mutating inputs to avoid surprises later on if the caller didn't expect the method to modify the input.

Reply:
It's not the `filter` that is mutating, it's the `historicalGift.donations = ...` assignment. That's because `historicalGifts[0]` and `filteredGifts[0]` refer to the same JavaScript object. You'll have to do something like `historicalGifts.map(historicalGift => ({ ...historicalGift, donations: historicalGift.donations.filter(...) })).filter(historicalGift => historicalGift.donations.length > 0)` to avoid mutating `historicalGifts`' donations.

Reply:
[cloneDeep](https://lodash.com/docs/#cloneDeep) seems to do the trick based on JSON outputs. Let me know what you think.

Reply:
The second `cloneDeep` (`cloneDeep(historicalGift.donations)`) shouldn't be necessary. Other than that it looks good!